src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventNotificationEmailManager.java

```diff
@@ -0,0 +1,175 @@
+                            .filter(DetailedEventBookingDTO -> bookingStatuses == null || bookingStatuses.contains(DetailedEventBookingDTO.getBookingStatus()))
+                            .map(DetailedEventBookingDTO::getUserBooked)
+                            .map(UserSummaryDTO::getId)
+                            .distinct().collect(Collectors.toList());
+        for (Long id : ids) {
+            try {
+                RegisteredUserDTO user = userAccountManager.getUserDTOById(id);
+                emailManager.sendTemplatedEmailToUser(user,
+                    emailManager.getEmailTemplateDTO(templateId),
+                    new ImmutableMap.Builder<String, Object>()
+                        .put("event.emailEventDetails", event.getEmailEventDetails() == null ? "" : event.getEmailEventDetails())
+                        .put("event", event)
+                        .build(),
+                    EmailType.SYSTEM);
+                log.debug(String.format("Sent email to user: %s %s, at: %s", user.getGivenName(), user.getFamilyName(), user.getEmail()));
+            } catch (NoUserException e) {
+                log.error(String.format("No user found with ID: %s", id));
+            } catch (ContentManagerException e) {
+                log.error("Failed to add the scheduled email sent time: ", e);
+            }
+        }
+    }
+
+    public void sendReminderEmails() {
+        // Magic number
+        Integer limit = 10000;
+        Integer startIndex = 0;
+        Map<String, List<String>> fieldsToMatch = Maps.newHashMap();
+        Map<String, Constants.SortOrder> sortInstructions = Maps.newHashMap();
+        Map<String, AbstractFilterInstruction> filterInstructions = Maps.newHashMap();
+        fieldsToMatch.put(TYPE_FIELDNAME, Collections.singletonList(EVENT_TYPE));
+        sortInstructions.put(DATE_FIELDNAME, Constants.SortOrder.DESC);
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime threeDaysAhead = now.plusDays(3);
+        DateRangeFilterInstruction
+            eventsWithinThreeDays = new DateRangeFilterInstruction(new Date(), Date.from(threeDaysAhead.toInstant()));
+        filterInstructions.put(ENDDATE_FIELDNAME, eventsWithinThreeDays);
+
+        try {
+            ResultsWrapper<ContentDTO> findByFieldNames = this.contentManager.findByFieldNames(
+                properties.getProperty(CONTENT_INDEX), ContentService.generateDefaultFieldToMatch(fieldsToMatch),
+                startIndex, limit, sortInstructions, filterInstructions);
+            for (ContentDTO contentResult : findByFieldNames.getResults()) {
+                if (contentResult instanceof IsaacEventPageDTO) {
+                    IsaacEventPageDTO event = (IsaacEventPageDTO) contentResult;
+                    String emailKey = String.format("%s@pre", event.getId());
+                    List<BookingStatus> bookingStatuses = Arrays.asList(BookingStatus.CONFIRMED, BookingStatus.ATTENDED);
```

Comment:
This was a request from Fergus, they shouldn't be marked as attended before an event but if for some reason an event manager does so accidentally then they will still receive the reminded. (Similar logic follows for sending the feedback email to confirmed / attended in case they don't set them to attended right away)

Reply:
OK sure, a short comment would be useful here to explain it.